### PR TITLE
Add add a utility for merging JSON fragments, and use it in run-container

### PR DESCRIPTION
**Pull request by brendandburns**

### CODE REVIEW

**Comment by erictune** [comment]
BTW, except for the api-version thing, you can accomplish the same thing with the `jq` command, e.g.
```
cat examples/guestbook/redis-slave-service.json  | jq .port=1234
{
  "id": "redisslave",
  "kind": "Service",
  "apiVersion": "v1beta1",
  "port": 1234,
  "containerPort": 6379,
  "labels": {
    "name": "redisslave"
  },
  "selector": {
    "name": "redisslave"
  }
}
```

**Comment by erictune** [comment]
@bgrant0607 you might care about this.

File: docs/kubectl.md [hunk 1]
```diff
@@ -913,6 +913,7 @@ Usage:
       --ns-path="/home/username/.kubernetes_ns": Path to the namespace info file that holds the namespace context to use for CLI requests.
   -o, --output="": Output format: json|yaml|template|templatefile
       --output-version="": Output the formatted object with the given version (default api-version)
+      --overrides="": An inline JSON override for the generated object.  If this is non-empty, it is parsed used to override the generated object
```
Comment (bgrant0607) [inline docs/kubectl.md]: Thanks.

File: pkg/kubectl/cmd/helpers.go [hunk 2]
```diff
@@ -26,7 +27,10 @@ import (
 	"strings"
 	"time"
 
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/runtime"
 	"github.com/golang/glog"
+	"github.com/imdario/mergo"
```
Comment (bgrant0607) [inline pkg/kubectl/cmd/helpers.go]: Does another PR add this to godeps?

File: docs/kubectl.md (hunk 1 above)
Comment (bgrant0607) [inline docs/kubectl.md]: Probably should document that apiVersion is required.

**Comment by bgrant0607** [comment]
@erictune This is something I asked for, as an escape hatch to head off feature creep. jq is cool, and has its uses, but 3 commands would be required to do this if using jq (`kubectl run-container --dry-run ... | jq ... | kubectl create -f ...`. If this turns out not to work well, then I agree we should rip it out and just recommend jq to users as a way to solve this problem.

**Comment by bgrant0607** [comment]
Also, I imagine mainly using this to specify things that are unset by run-container rather than modifying things that it does set. That seems hard with jq.

File: docs/kubectl.md (hunk 1 above)
Comment (brendandburns) [inline docs/kubectl.md]: done.

File: pkg/kubectl/cmd/helpers.go (hunk 2 above)
Comment (brendandburns) [inline pkg/kubectl/cmd/helpers.go]: No, it's actually already there.  looks like @deads2k added it with the config changes:

https://github.com/GoogleCloudPlatform/kubernetes/tree/master/Godeps/_workspace/src/github.com/imdario/mergo

**Comment by brendandburns** [comment]
Comments addressed, ptal.

Thanks
--brendan

**Comment by bgrant0607** [comment]
LGTM. Waiting for Travis to get unstuck.

**Comment by smarterclayton** [comment]
I'm lazy and usually just merge the follow on one (so you only get one merge commit).  GitHub will detect that this is merged and close it automatically.

**Comment by bgrant0607** [comment]
Thanks for the tip, @smarterclayton. Will do.

**Pull request merged**